### DISCUSSION TITLE
How are people bulk adding Tags?

### DISCUSSION THREAD

**Discussion by lewsut** [frangoteam/FUXA#970] [Q&A]
I have around 200 tags and a bunch of these exist within MQTT payloads. I have looked at the device export and adding regular tags is annoying but not impossible, however creating a MQTT payload, within the CSV file with generated t_ GUIDs is just messy and laborious and very open to error. 

Am I missing something? Currently it's easier to manually type them in ( one at a time) to the GUI and that's seems silly.

Thanks for any ideas

**Comment by unocelli** [comment]
Hi, the t_GUIDs is not a constraint, you can also put a simple index 1..n, it is important that it is unique on all tags in the project

**Reply by lewsut** [reply]
T_GUID was just and example, however tags' names being unique would be just as good? Basically the setup for MQTT payloads is unworkable for my simple brain having to use an abstract index in place of a readable name.

Currently I'm inclined to change my microcontroller code to MQTT raw single topics per data point as it's a lot less messy in terms of FUXA import export.

It's good fun otherwise, cheers.

**Reply by unocelli** [reply]
use names to identify tags, perhaps it was done in the 1970s

**Reply by lewsut** [reply]
the backend is obviously GUIDed however the user interface , import and export of all the SCADA HMIs and PLCS I use day to day have "pretty" names usually unique throughout the app. In the mid 80s I was programming PLCs with cassette tapes ;)

**Reply by lewsut** [reply]
![image](https://github.com/frangoteam/FUXA/assets/25065897/9b46f2f5-81f2-42ab-93cd-f391d0afb0b3)
An example from RS500, nice readable tag names, the ID aspect never come into play is all obsurated during imports and exports 

This is very similar in Ignition, Panorama, Factorylink, Crimson and so on .

**Reply by unocelli** [reply]
how is the case if I want to export to change the name of some tags and then import it again. I have to configure again all bind to the GUI?

**Reply by lewsut** [reply]
Yes, I have built a few automatic GUI importers (standard object mimics) , however the vast majority of the use for tag import / export if for backed tag creation and edit, rarely would you be adding and editing tags that are bound to a GUI.  

You just want some way to get the tags into tegh backend so you can use them, easily, then bind .

**Reply by unocelli** [reply]
then imagine you have 100 tags that you have already associated with GUI controls, now the names of the tags are changed to your schema (e.g. the prefix for change of position). then you have to go and change on all gui.

**Reply by lewsut** [reply]
I work typically with 10,000 - 100,000 tags ;) many modern scada package have friendly name tag import/ export tools. The older ones don't but we hardly touch them now, I honestly can't think of a time changing a few tag names caused anything destructive, certainly not since Factory Link ...

**Reply by unocelli** [reply]
you don't have reply to my question!...do you make the binding with the graphical component (gauge), via the tag name? and when you edit or rename the name do you search the entire project where you made the binding to change it?

**Comment by robbudge** [comment]
I’m thinking a Python script could open the JSON and then add the contents of a CSV to a specific device with default settings and a random uuid. 

Automatic import has caused problems with other platforms in the past as you normally get EVERY TAG. 
often far more than you actually use in the HMI and this is just a waste of resources.

**Reply by lewsut** [reply]
Perhaps but a well thought out CSV file is easy enough to filter and ignore the chaff.

**Comment by robbudge** [comment]
I am currently experimenting with Codesys and mqtt, all is going well but creating each tag in Fuxa one by one is slow.
I performed a JSON tag export from Fuxa and built  very simple Python script that connects  to the broker and builds a JSON file that can be pasted into the  exported file.
Its not fancy or pretty, but works for my application and maybe be useful to someone else.

[MQTT_to_Fuxa.zip](https://github.com/frangoteam/FUXA/files/13547305/MQTT_to_Fuxa.zip)

[Video](https://youtu.be/7gZN1sgP0Rc)
